SharkCage/CageManager/CageLabeler.h

```diff
@@ -173,22 +235,63 @@ bool CageLabeler::ShowCageWindow()
 	return true;
 }
 
-void DisplayTokenInCageWindow(HWND *hwnd)
+static bool DisplayTokenInCageWindow(HWND *hwnd)
 {
 	std::wcout << L"starting display image" << std::endl;
+	
+	const std::string token_string(app_tokens.begin(), app_tokens.end());
+
+	std::string decoded_image = base64_decode(token_string);
+
+	DWORD image_size = decoded_image.length();
+
+	HGLOBAL h_memory = ::GlobalAlloc(GMEM_MOVEABLE, image_size);
+	if (h_memory == NULL)
+	{
+		std::cout << "Failed to allocate memory for token image. Error " << ::GetLastError() << std::endl;
+		return false;
+	}
+
+	LPVOID p_image = ::GlobalLock(h_memory);
+	if (p_image == NULL)
+	{
+		std::cout << "Failed to allocate memory for token image. Error " << ::GetLastError() << std::endl;
+		return false;
+	}
+
+	::memcpy(p_image, decoded_image.c_str(), image_size);
```

Comment:
`std::copy` might be a better alternative

Reply:
Tried it with `std::copy` but wasn't successful to copy from a `std::string` to `LPVOID`. Tried a few thinks but without any working solution 😞

Reply:
right, `std::copy` was probably a stupid suggestion. but we can use the C++ version of `memcpy` from the `cstring` header: http://en.cppreference.com/w/cpp/string/byte/memcpy